### PR TITLE
Fix sorting of caret

### LINKED ISSUE #237: Incorrectly rewrite `/[a^]/` to `/[^a]/`

### PR DESCRIPTION
**Pull request by fisker**
Fixes #237

### CODE REVIEW

File: src/optimizer/__tests__/optimizer-integration-test.js [hunk 1]
```diff
@@ -35,6 +35,32 @@ describe('optimizer-integration-test', () => {
...
+    expect(optimizer.optimize(original).toString()).toBe(optimized.toString());
+
+    original = /[^-]/;
+    optimized = /[^-]/;
+
+    expect(optimizer.optimize(original).toString()).toBe(optimized.toString());
+
+    // TODO: Fix this
+    original = /[-^]/;
+    optimized = /[^-]/;
```
Comment (fisker) [inline src/optimizer/__tests__/optimizer-integration-test.js:61]: This still broken, I'm not sure how to fix it. Maybe add some unsortable case?

Comment (DmitrySoshnikov) [reply src/optimizer/__tests__/optimizer-integration-test.js:61]: Yeah, this seems a regression, current version doesn't produce `/[^-]/` in this case. I think we should treat `^` as unsortable generically.

Comment (DmitrySoshnikov) [reply src/optimizer/__tests__/optimizer-integration-test.js:61]: Actually, maybe returning `return Number.MAX_SAFE_INTEGER` both for `-` and `^` would help?

File: src/optimizer/transforms/char-class-classranges-merge-transform.js [hunk 2]
```diff
@@ -95,10 +95,13 @@ function sortCharClass(a, b) {
 function getSortValue(expression) {
   if (expression.type === 'Char') {
     if (expression.value === '-') {
-      return Infinity;
+      return Number.MAX_SAFE_INTEGER;
+    }
+    if (expression.value === '^') {
+      return Number.MAX_SAFE_INTEGER - 1;
```
Comment (DmitrySoshnikov) [inline src/optimizer/transforms/char-class-classranges-merge-transform.js:101]: Should it also be `Number.MAX_SAFE_INTEGER`? In this case `/[-^]/` would stay the same.

File: src/optimizer/__tests__/optimizer-integration-test.js (hunk 1 above)
Comment (fisker) [reply src/optimizer/__tests__/optimizer-integration-test.js:61]: No, `[-^]` `[^-]` has to be different, but current solution will break one of them, whatever we choose the order it. We have to find a different solution.

Comment (DmitrySoshnikov) [reply src/optimizer/__tests__/optimizer-integration-test.js:61]: I see – yes, then maybe putting them at the very end instead or adding extra slashes post-sorting?

**Pull request closed without merging**